oshmem/shmem/c/shmem_ptr.c

```diff
@@ -19,18 +19,53 @@
 #include "oshmem/shmem/shmem_api_logger.h"
 
 #include "oshmem/runtime/runtime.h"
+#include "oshmem/mca/memheap/memheap.h"
+#include "oshmem/mca/memheap/base/base.h"
+
 
 #if OSHMEM_PROFILING
 #include "oshmem/include/pshmem.h"
 #pragma weak shmem_ptr = pshmem_ptr
 #include "oshmem/shmem/c/profile/defines.h"
 #endif
 
-void *shmem_ptr(const void *ptr, int pe)
+void *shmem_ptr(const void *dst_addr, int pe)
 {
-    SHMEM_API_VERBOSE(10,
-                      "*************** WARNING!!! NOT SUPPORTED FUNCTION **********************\n"
-                      "shmem_ptr() function is available only on systems where ordinary memory loads\n"
-                      "and stores are used to implement OpenSHMEM put and get operations.");
-    return 0;
+    ompi_proc_t *proc;
+    sshmem_mkey_t *mkey;
+    int i;
+    void *rva;
+
+    RUNTIME_CHECK_INIT();
+    RUNTIME_CHECK_PE(pe);
+    RUNTIME_CHECK_ADDR(dst_addr);
+
+    /* process can access its own memory */
+    if (pe == oshmem_my_proc_id()) {
+        return (void *)dst_addr;
+    }
+
+    /* The memory must be on the local node */
+    proc = oshmem_proc_group_find(oshmem_group_all, pe);
```

Comment:
don't think we should restrict the access to "local nodes". if the spml has some magic to be able to access to remote nodes as well, need to use it.